### PR TITLE
Always use most recent term

### PR DESCRIPTION
**Pull request by kevinjihwanlee**
Previously, we retrieved data from Firestore using a hard-coded term ID. After this ticket, Serif should now determine upon load what the latest term should be, and use that term value to load the appropriate data. This required some rewriting of the Redux architecture by including action enhancers.

Note that in this PR, the term ID displayed on `TopBar` is still hard-coded. This will be changed once the Hooks conversion PR is merged, so that we can use `useSelector` without having to write an entire container that will eventually get removed.

### CODE REVIEW

File: src/effects/browse.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import { db } from 'util/firebase';
-import { CURRENT_TERM } from 'util/constants';
 
-const currentTermDoc = db.collection('terms').doc(CURRENT_TERM);
-
-export function fetchSchools() {
+export function fetchSchools(currentTerm) {
+  const currentTermDoc = db.collection('terms').doc(currentTerm);
```
Comment (Joonpark13) [inline src/effects/browse.js]: This is duplicated three times - could be a function instead

File: src/effects/globals.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable import/prefer-default-export */
+import { db } from 'util/firebase';
+
+export function fetchCurrentTerm() {
+  return db.collection('terms').get()
+    .then(querySnapshot => querySnapshot.docs.reduce(
```
Comment (Joonpark13) [inline src/effects/globals.js]: Can you `orderBy` instead? https://firebase.google.com/docs/firestore/query-data/order-limit-data

File: src/reducers/browse.js [hunk 3]
```diff
@@ -92,7 +94,7 @@ function browse(state = initialBrowseState, action) {
       return loop(
         state.set('isFetching', true),
         Cmd.run(fetchSubjects, {
-          args: [action.schoolId],
+          args: [action.currentTerm.get('id'), action.schoolId],
```
Comment (Joonpark13) [inline src/reducers/browse.js]: We don't need the whole current term object. The enhancer should just provide the current term id.

File: src/effects/__mocks__/globals.js [hunk 4]
```diff
@@ -1,2 +1,2 @@
 /* eslint-disable import/prefer-default-export */
-export const CURRENT_TERM = '4760';
+export const fetchCurrentTerm = () => {};
```
Comment (Joonpark13) [inline src/effects/__mocks__/globals.js:2]: This file can just be deleted, right?

Comment (kevinjihwanlee) [reply src/effects/__mocks__/globals.js:2]: This is so I can run the reducer test for `globals` properly

File: src/effects/globals.js (hunk 2 above)
Comment (kevinjihwanlee) [reply src/effects/globals.js]: Done

File: src/reducers/browse.js (hunk 3 above)
Comment (kevinjihwanlee) [reply src/reducers/browse.js]: Done

File: src/actions/action-enhancers.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { currentTermSelector } from 'selectors';
+
+export const ENHANCE_WITH_CURRENT_TERM_ID = Symbol('ENHANCE WITH CURRENT TERM');
+const termEnhancer = {
```
Comment (Joonpark13) [inline src/actions/action-enhancers.js]: `termIdEnhancer`?

**Review by Joonpark13 — APPROVED** [review]

**Pull request merged**